src/reason-parser/reason_declarative_lexer.mll

```diff
@@ -320,6 +320,10 @@ let operator_chars =
   ['!' '$' '%' '&' '+' '-' ':' '<' '=' '>' '?' '@' '^' '|' '~' '#' '.'] |
   ( '\\'? ['/' '*'] )
 
+let dotsymbolchar =
+  ['!' '$' '%' '&' '*' '+' '-' '/' ':' '=' '>' '?' '@' '^' '|']
```

Comment:
What do y'all think about also supporting alphanumeric characters in the let bindings? I think this is a great opportunity for us to "make it our own" -- ocaml's syntax tends to be more terse and rely on sigils, whereas (imo) reason goes more for descriptive names.
For example, I would much rather be able to define 
```
let (let+async) = ...
```
than
```
let (let+<==>&) = ...
```

```suggestion
  ['!' '$' '%' '&' '*' '+' '-' '/' ':' '=' '>' '?' '@' '^' '|' 'a'-'z' 'A'-'Z' '_' '0'-'9']
```

Reply:
That's a good suggestion -- I'm happy to add that if it doesn't cause parser conflicts

Reply:
That's a great idea @jaredly. And at the same time it's compatible with the existing OCaml toolchain/pattern.

Reply:
Can this still be printed to .ml with same AST? Or refmt will do the AST transformation on its own? It seems that there is no way to express `let+opt` in OCaml source according to [documentation on binding operators](http://caml.inria.fr/pub/distrib/ocaml-4.08/ocaml-4.08-refman.html#s%3Abinding-operators).

Reply:
@Lupus the ml printer will happily print it, it just won't be parseable :)
We could easily add a rewrite step, to encode e.g. `let+opt` into something valid, if that's a use case that people want. I believe that currently, due to bugs in the core ocaml printer, there are already cases where unparseable code is output.

Reply:
And as you say, the other option is to use the "backport" logic when printing to .ml if the extended character range is used. (note that the binary ast format, which is used by dune and bsb under normal reason development, will happily accept the extended character range)

Reply:
It would be good to at least get some kind of loose commitment that the compiler will not one day end up rejecting alphanumeric identifiers in this portion of the AST.